tests/fused_attention_stablehlo_test.py

```diff
@@ -107,55 +154,50 @@ def sdpa_train_ref(query: Array,
             bias: Optional[Array] = None,
             mask: Optional[Array] = None,
             scale: float = 0.5,
-            is_causal_mask: bool = False,
+            mask_type: MaskType = MaskType.NO_MASK,
             dropout_rate: float = 0.1) -> Array:
   out_ref, sdpa_vjp_ref = jax.vjp(
-    partial(sdpa_ref, scale=scale, is_causal_mask=is_causal_mask, dropout_rate=dropout_rate),
+    partial(
+      sdpa_ref, scale=scale, mask_type=mask_type, dropout_rate=dropout_rate),
     query, key, value, bias, mask)
-  query_grad_ref, key_grad_ref, value_grad_ref, _, _ = sdpa_vjp_ref(grad)
+  query_grad_ref, key_grad_ref, value_grad_ref, bias_grad_ref, _ = sdpa_vjp_ref(grad)
+  if bias is not None and len(bias.shape) == 3:
+    return out_ref, (query_grad_ref, key_grad_ref, value_grad_ref, bias_grad_ref)
   return out_ref, (query_grad_ref, key_grad_ref, value_grad_ref)
 
 class DotProductAttentionTest(jtu.JaxTestCase):
   def setUp(self):
-    # TODO(Cjkkkk): Tests fail when the cuDNN constraints check fails, or
-    #  even when it passes: https://github.com/google/jax/issues/20438
-    self.skipTest('Failing with various errors')
-    return
-
+    super().setUp()
     if jax.device_count() < 4:
       self.skipTest("Requires more than 4 devices.")
     try:
       cudnn_version = check_cudnn_version()
+      check_compute_capability((80, 90))
     except RuntimeError as e:
       self.skipTest(str(e))
       return
     if cudnn_version < 8904:
-      self.skipTest('Requires >= cuDNN 8.9.4')
+      self.skipTest("Requires >= cuDNN 8.9.4")
 
   @jtu.sample_product(
       batch_size=[4],
-      seq_len=[256, 1024],
+      seq_len=[1024],
       num_heads=[8],
       head_dim=[64, 128],
       use_bias=[False, True],
-      use_mask=[False, True],
-      is_causal_mask=[False],
+      mask_type=[MaskType.NO_MASK],
```

Comment:
Does it make sense to test with different masks here?

Reply:
Added causal. padding is tested in `test_sdpa_var_seq`